### PR TITLE
Borrowck error reporting cleanup

### PR DESCRIPTION
**Pull request by matthewjasper**
* Don't show variables created by desugarings in borrowck errors
* Move "conflict error" reporting to it's own module, so that `error_reporting` contains only common error reporting methods.
* Remove unused `ScopeTree` parameter.

r? @pnkfelix

### CODE REVIEW

File: src/librustc_mir/borrow_check/error_reporting.rs [hunk 1]
```diff
@@ -1764,11 +1764,11 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
     fn append_local_to_string(&self, local_index: Local, buf: &mut String) -> Result<(), ()> {
         let local = &self.mir.local_decls[local_index];
         match local.name {
-            Some(name) => {
-                buf.push_str(&name.to_string());
+            Some(name) if !local.from_compiler_desugaring() => {
```
Comment (pnkfelix) [inline src/librustc_mir/borrow_check/error_reporting.rs]: can you update the doc-comment for this method to reflect its behavior more accurately? Something like:

```
/// Appends end-user visible description of the `local` place to `buf`. If `local` doesn't have
/// a name, or if its name was generated by the compiler, then `Err` is returned and `buf` is left unmodified.
```

File: src/test/ui/nll/dont-print-desugared.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Test that we don't show variables with from internal desugarings
+
+// edition:2018
```
Comment (pnkfelix) [inline src/test/ui/nll/dont-print-desugared.rs]: this is only testing behavior for 2018 edition.

Can you add a similar test that checks the 2015 edition as well? (Or remove the "edition" mark from this file...?) Or is there some reason it makes sense to only test this against the 2018 edition? (Maybe I missed some new policy regarding testing in terms of whether to encode an edition or not.)

**Comment by pnkfelix** [comment]
r=me once review nits above are addressed.

File: src/test/ui/nll/dont-print-desugared.rs (hunk 2 above)
Comment (matthewjasper) [reply src/test/ui/nll/dont-print-desugared.rs]: `async fn` is >=2018 edition only

Comment (matthewjasper) [reply src/test/ui/nll/dont-print-desugared.rs]: I've split the `async fn` case into its own test to maybe make this more obvious.

**Pull request merged**